### PR TITLE
quill: new wrap

### PR DESCRIPTION
**Pull request by odygrd**
Quill is a cross-platform low latency C++ logging library. The project can be found [here](https://github.com/odygrd/quill)

### CODE REVIEW

**Comment by jpakkane** [comment]
You need to give the version number a "wrapdb version is", it should be `4.2.0-1`.

File: subprojects/packagefiles/quill/meson.build [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+quill_dep = declare_dependency(include_directories : inc_dirs,
+                               dependencies : [dependency('threads')],
+                               compile_args : interface_compile_args)
+
+pkg_mod = import('pkgconfig')
+pkg_mod.generate(
+    name : 'quill',
+    filebase : 'quill',
+    description : 'Asynchronous Low Latency C++ Logging Library',
+    version : '4.2.0',
```
Comment (benoit-pierre) [inline subprojects/packagefiles/quill/meson.build:22]: This should already default to the project version if unspecified.

Comment (odygrd) [reply subprojects/packagefiles/quill/meson.build]: thanks, i have removed the version

File: subprojects/packagefiles/quill/meson.build [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+inc_dirs = include_directories('quill/include')
+
+# Conditional compiler arguments for specific compiler
+cpp = meson.get_compiler('cpp')
+if cpp.get_id() == 'gcc' or cpp.get_id() == 'clang'
+    interface_compile_args = ['-Wno-gnu-zero-variadic-macro-arguments']
+else
+    interface_compile_args = []
+endif
```
Comment (benoit-pierre) [inline subprojects/packagefiles/quill/meson.build:11]:
```suggestion
interface_compile_args = meson.get_compiler('cpp').get_supported_arguments('-Wno-gnu-zero-variadic-macro-arguments')
```

File: subprojects/packagefiles/quill/meson.build [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+quill_dep = declare_dependency(include_directories : inc_dirs,
+                               dependencies : [dependency('threads')],
+                               compile_args : interface_compile_args)
+
+pkg_mod = import('pkgconfig')
+pkg_mod.generate(
+    name : 'quill',
+    filebase : 'quill',
+    description : 'Asynchronous Low Latency C++ Logging Library',
+    subdirs : 'quill/include'
```
Comment (benoit-pierre) [inline subprojects/packagefiles/quill/meson.build:22]: Don't forget to actually install the required headers with `install_headers(…)`.

Comment (odygrd) [reply subprojects/packagefiles/quill/meson.build:17]: thanks, would this work ?

install_subdir('quill/include', install_dir : 'include/quill')

Comment (benoit-pierre) [reply subprojects/packagefiles/quill/meson.build:17]: If you go this way, make sure the install tag is correct, and the `includedir` option is honored, this should work (please double check):
```meson
install_subdir('quill/include', install_dir: get_option('includedir') / 'quill', install_tag: 'devel', strip_directory: true)
```

**Review by benoit-pierre — APPROVED** [review]

File: subprojects/packagefiles/quill/meson.build [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+inc_dirs = include_directories('quill/include')
+
+# Conditional compiler arguments for specific compiler
+interface_compile_args = meson.get_compiler('cpp').get_supported_arguments('-Wno-gnu-zero-variadic-macro-arguments')
+
+quill_dep = declare_dependency(include_directories : inc_dirs,
+                               dependencies : [dependency('threads')],
+                               compile_args : interface_compile_args)
+
+install_subdir('quill/include/quill', install_dir: get_option('includedir') / 'quill', install_tag: 'devel', strip_directory: true)
```
Comment (jpakkane) [inline subprojects/packagefiles/quill/meson.build:12]: A simpler version of this should be

    install_subdir('quill/include/quill', install_dir: get_option('includedir')', install_tag: 'devel')

**Review by benoit-pierre — APPROVED** [review]

**Pull request merged**